AutoRest/Generators/NodeJS/NodeJS/ClientModelExtensions.cs

```diff
@@ -606,6 +623,26 @@ private static string SerializePrimaryType(this PrimaryType primary, IScopeProvi
                                 .Outdent()
                                 .AppendLine("}").ToString();
             }
+            else if (primary == PrimaryType.TimeSpan)
+            {
+                if (isRequired)
+                {
+                    builder.AppendLine("if(!{0} || !moment.isDuration({0})) {{", objectReference);
+                    builder = ConstructValidationCheck(builder, requiredTypeErrorMessage, objectReference, primary.Name);
+                    builder = ConstructBasePropertyCheck(builder, valueReference);
+                    return builder.AppendLine("{0} = {1}.toISOString();", valueReference, objectReference).ToString();
+                }
+
+                builder.AppendLine("if ({0}) {{", objectReference)
+                         .Indent()
+                         .AppendLine("if !(moment.isDuration({0})) {{",
```

Comment:
I followed the pattern of the other primary types (DateTime, Date, ByteArray, etc) which follow this same pattern.